docs/network-policy.md

```diff
@@ -0,0 +1,122 @@
+# Cluster Network Policy
+
+ClusterNetworkPolicy is a specification of how workloads within a cluster
+communicate with each other and other external endpoints.
+The ClusterNetworkPolicy is supposed to aid cluster-admins to determine
+the security policy for the cluster, unlike K8s NetworkPolicy, which is
+aimed towards developers to secure their apps.
+
+## The ClusterNetworkPolicy resource
+
+An example ClusterNetworkPolicy might look like this:
+```
+apiVersion: security.antrea.tanzu.vmware.com/v1beta1
+kind: ClusterNetworkPolicy
+metadata:
+  name: test-cnp
+spec:
+    priority: 1
+    appliedTo:
+    - podSelector:
+        matchLabels:
+          role: db
+    - namespaceSelector:
+        matchLabels:
+          env: prod
+    ingress:
+    - action: Allow
+      from:
+      - podSelector:
+          matchLabels:
+            role: db
+      - podSelector:
+          matchLabels:
+            role: db
+        namespaceSelector:
+          matchLabels:
+            role: db
+      ports:
+      - protocol: TCP
+        port: 8080
+        name: httpPort
+    egress:
+    - action: Drop
+      to:
+      - ipBlock:
+        cidr: 10.0.10.0/24
+      ports:
+      - protocol: TCP
+        port: 5978
+```
+
+**spec**: The ClusterNetworkPolicy `spec` has all the information needed to
+define a cluster-wide security policy.
+
+**appliedTo**: The `appliedTo` field specifies the grouping criteria of Pods to
+which the policy applies to. Pods can be selected cluster-wide using
+`podSelector`. If set with a `namespaceSelector`, all Pods from Namespaces
+selected by the namespaceSelector will be selected. Specific Pods from
+specific Namespaces can be selected by using a combination of
+`podSelector` and `namespaceSelector` in the same peer. IPBlock is not allowed
+to be set in the `appliedTo` field.
+In the example, the policy applies to Pods, which either match the labels
+"role=db" in all the Namespaces, or are from Namespaces which match the
+labels "env=prod".
+
+**priority**: The `priority` field determines the relative priority of the policy
+among all ClusterNetworkPolicies in the given cluster. This field is mandatory.
+A lower priority value indicates higher precedence. Priority values can range
+from 1-10000.
```

Comment:
Can we recommend users to set priorities between 1-100 to start with for v0.8? Current implementation can handle 1-10000, but putting up huge priority values tend to overflow the last priorityZone.

Reply:
done